pkg/services/publicdashboards/api/api.go

```diff
@@ -239,3 +250,15 @@ func toJsonStreamingResponse(features *featuremgmt.FeatureManager, qdr *backend.
 
 	return response.JSONStreaming(statusCode, qdr)
```

Comment:
I think this is the flow I would think of for this method (includes inline comments I left on this review as well). I added here because they're not really organized well.. so hopefully this better articulates what I'm thinking from the top down.

![image](https://user-images.githubusercontent.com/1669620/186550272-be9ee9d4-76bb-4131-a83e-ad6d6ef4b37f.png)

Reply:
I made some changes and answered some comments. Let's talk about this with the new structure in mind.